Source/Graphic/HCRaster+File.c

```diff
@@ -0,0 +1,348 @@
+            if (!binary) {
+                fprintf(file, "%i %i %i ", (int)r, (int)g, (int)b);
+            }
+            else {
+                fwrite(&r, sizeof(HCByte), 1, file);
+                fwrite(&g, sizeof(HCByte), 1, file);
+                fwrite(&b, sizeof(HCByte), 1, file);
+            }
+        }
+        if (!binary) {
+            fprintf(file, "\n");
+        }
+    }
+    
+    // Cleanup
+    fclose(file);
+}
+
+HCRasterRef HCRasterCreateByLoadingBMP(const char* path) {
+    // BMP (Windows V3)
+    // Offset    Size    Description
+    // 0         2       the magic number used to identify the BMP file: 0x42 0x4D (Hex code points for B and M in big-endian order)
+    // 2         4       the size of the BMP file in bytes
+    // 6         2       reserved; actual value depends on the application that creates the image
+    // 8         2       reserved; actual value depends on the application that creates the image
+    // 10        4       the offset, i.e. starting address, of the byte where the bitmap data can be found.
+    // 14        4       the size of this header (40 bytes)
+    // 18        4       the bitmap width in pixels (signed integer).
+    // 22        4       the bitmap height in pixels (signed integer).
+    // 26        2       the number of color planes being used. Must be set to 1.
+    // 28        2       the number of bits per pixel, which is the color samplesPerPixel of the image. Typical values are 1, 4, 8, 16, 24 and 32.
+    // 30        4       the compression method being used. See the next table for a list of possible values.
+    // 34        4       the image size. This is the size of the raw bitmap data (see below), and should not be confused with the file size.
+    // 38        4       the horizontal resolution of the image. (pixel per meter, signed integer)
+    // 42        4       the vertical resolution of the image. (pixel per meter, signed integer)
+    // 46        4       the number of colors in the color palette, or 0 to default to 2n.
+    // 50        4       the number of important colors used, or 0 when every color is important; generally ignored.
+    // 51        4*w*h   pixel data in XXRRGGBB uint32 format
+    
+    // Open file for reading
+    FILE* f = fopen(path, "r");
+    if (f == NULL) {
+        return NULL;
+    }
+    
+    // Define header data
+    uint16_t magicNumber = 0x4D42;
+    uint32_t fileSize = 0;
+    uint16_t reserved0 = 0;//0x4D41;
+    uint16_t reserved1 = 0;//0x5454;
+    uint32_t dataOffset = 54;
+    uint32_t infoHeaderSize = 40;
+    int32_t width = 0;
+    int32_t height = 0;
+    //uint16_t colorPlanes = 1;
+    //uint16_t bitsPerPixel = 32;
+    //uint32_t compression = 0;
+    //uint32_t dataSize = width * height * bitsPerPixel / 8;
+    //int32_t horizontalResolution = 2835;
+    //int32_t verticalResolution = 2835;
+    //uint32_t paletteColorCount = 0;
+    //uint32_t importantPaletteColorCount = 0;
```

Comment:
Do you want these here?

Reply:
Yes, they are related to the TODO: on 206